src/main/java/com/xiaomi/infra/pegasus/security/KerberosProtocol.java

```diff
@@ -41,32 +48,21 @@
   // request. The JAAS framework defines the term "subject" to represent the source of a request. A
   // subject may be any entity, such as a person or a service.
   private Subject subject;
-  // The LoginContext class provides the basic methods used to authenticate subjects, and provides a
-  // way to develop an application independent of the underlying authentication technology
-  private LoginContext loginContext;
   private String serviceName;
   private String serviceFqdn;
+  private String keyTab;
+  private String principal;
+  final int CHECK_TGT_INTEVAL_SECONDS = 10;
```

Comment:
from 
> 公司内部的这个renew till时间是10年，所以我们平时测试的话，确实不容易看出问题

I understand the refresh is not necessary in some case, but the 10s task will Increase client load, so better add a config to whether enable auto-refresh?

Reply:
The refresh is necessary. Because the renew till time maybe be changed someday. And I think the timer runs every 10 seconds will not cost too much